generator/python/build.gradle

```diff
@@ -49,39 +50,43 @@ task generateClient(type: GradleBuild) {
         
         def clientOutputDirPerVersion = clientOutputDir + "/" + parsedApiName;
         def specPath = it.path;
-        def packageBase =  criteoApiVersion; 
-        if (packageBase.matches("[0-9].*"))
-             packageBase= "v" + packageBase;
+        def versionCategory = criteoApiVersion == 'preview' ? 'preview' : 'stable'
+  
+        def versionInNamespace =  criteoApiVersion
+        if (versionInNamespace.matches("[0-9].*"))
+             versionInNamespace= "v" + versionInNamespace
 
-        packageBase = "${criteoPackage}.${criteoService}.${packageBase}".replace('.', nameSeparator).toString(); // example : criteo_api_marketingsolutions_v2021_10
-        def packageVersion = "${criteoApiVersion}.${generatorVersion}.${getFormattedDate()}"; // example : 2021_10.1.211110
-
-        def generateTask = task("openApiGenerate_python_" + parsedApiName, type: GenerateTask.class) {
+        def namespaceBase = "${criteoPackage}.${criteoService}.${versionInNamespace}".replace('.', nameSeparator).toString(); // example: criteo\api\marketingsolutions\v2021_10
+        
+        def artifactVersion = versionCategory == 'preview' ? '0' : criteoApiVersion.replaceAll("_",".").toString()
+        artifactVersion += ".${generatorVersion}.${getFormattedDate()}".toString() // example : 2021.01.0.211110 for stable , 0.0.211110 for preview
+ 
+        def generateTask = task("openApiGenerate_" + language + "_" + parsedApiName, type: GenerateTask.class) {
                 generatorName = "python"
                 templateDir = "$resourcesDir/templates/".toString()
                 inputSpec = specPath
                 outputDir = clientOutputDirPerVersion
                 removeOperationIdPrefix = true
                 generateApiTests = false
                 generateModelTests = false
-                gitUserId='criteo'
-                gitRepoId='criteo-api-python-sdk'
+                gitUserId = 'criteo'
+                gitRepoId = "criteo-api-${criteoService}-${versionCategory}-sdk".toString()
                 configOptions = [
-                        apiVersion              : parsedJson.info.version,
-                        packageName             : "${packageBase}".toString(),
-                        invokerPackage          : 'Criteo\\SDK',
-                        packageUrl              : 'https://github.com/criteo/criteo-api-python-sdk',
-                        hideGenerationTimestamp : 'true',
-                        artifactId              : "criteo-api-${criteoService}-sdk-${technologyStack}".toString(),
-                        packageVersion          : "${packageVersion}".toString(), // v2021_10.1.211109
                         groupId                 : 'criteo',
-                        version                 : '1.0.0',
+                        apiVersion              : "${artifactVersion}".toString(),
+                        artifactId              : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(),
```

Comment:
same as before, the version category should not be included in the package name

Reply:
We will update the story and take these latest decision into account in next commit